pulp_container/app/authorization.py

```diff
@@ -174,15 +178,17 @@ def has_push_permissions(self, path):
             distribution = ContainerDistribution.objects.get(base_path=path)
         except ContainerDistribution.DoesNotExist:
             # distribution does not exist, need to create a new one
-            if not self.user.has_perm("container.add_containerdistribution"):
-                return False
             namespace = path.split("/")[0]
             return NamespacePermissionsChecker.has_permissions(
-                namespace, self.user, "container.manage_namespace_distributions"
+                namespace, self.user, "container.namespace_add_containerdistribution"
             )
         # this is an existing distribution. User should have model or object perms
-        if self.user.has_perm("container.push_containerdistribution") or self.user.has_perm(
-            "container.push_containerdistribution", distribution
+        if (
```

Comment:
we should probably add model perms of `container.namespace_push_containerdistribution` to this logical statement